### ISSUE TITLE
Windows Long Path issue breaks display of rendered file

### ISSUE THREAD

**Issue by finagleartistry** [sebthom/previewer-eclipse-plugin#10]
On Windows 10 / Eclipse 2023-12 / OpenJDK 17/21, rendering fails if the source path leads to an overlong path to the cache file.
WebView2 will display "File not found",  Internet Explorer will display "Can't reach this page", even though the rendered files are present in the cache. The rendered_content.html files are valid and can be opened in Chrome or Edge directly.

This can be examined by placing an identical test Markdown file in 2 folders and increasing the folder name until render display fails. This can be narrowed down to a single character changing the behavior. This seems to put the rendered cache file path over the Windows path limit.

Note that Windows Long Path support is enabled in both policy and registry.

A: Source path: 114 characters. Rendered file path: 258 characters - OK
B: Source path: 115 characters. Rendered file path: 262 characters - Not Found

**Comment by sebthom** [comment]
why are you getting so long paths? Mine are 210 chars max.
```
file:///C:/Users/User1/AppData/Local/Temp/de.sebthom.eclipse.previewer/render_cache_files/RDpcd29ya3NwYWNlc1xwcm9qZWN0c1xwcmV2aWV3ZXItZWNsaXBzZS1wbHVnaW5cQ09ERV9PRl9DT05EVUNULm1k/rendered_content.html
```

**Comment by finagleartistry** [comment]
As far as I understand the code (not being a Java coder) - they are formed from cache dir location + base64 encode of the source path, so the length of the rendered file path is a function of the length of the source file path.

The source file paths are quite long because there is some depth before the root of the project is reached and we have "meaningful" filenames.

My username is 7 characters, which makes the cache prefix slightly longer than your example, but I don't think that is it.

Are you starting with source file paths that  are 115 characters and then getting 210 character rendered file paths?

**Comment by sebthom** [comment]
you were actually right. I did use base64 for part of the cache file paths. I fixed that now f3b1a723b4190e251b75ae0c907ecb400ae517c2

Thanks for reporting!

**Comment by finagleartistry** [comment]
Thanks for fixing!
